code/code/sys/connect.cc

```diff
@@ -2881,77 +2879,65 @@ int Descriptor::sendLogin(const sstring& arg) {
   if (my_arg.empty())
     return DELETE_THIS;
   else if (my_arg == "?") {
-    writeToQ(
+    constexpr std::array messages = {
       "Accounts are used to store all characters belonging to a given "
-      "person.\n\r");
-    writeToQ(
-      "One account can hold multiple characters.  Creating more than one "
-      "account\n\r");
-    sprintf(buf,
-      "for yourself is a violation of %s multiplay rules and will lead to\n\r",
-      MUD_NAME);
-    writeToQ(buf);
-    writeToQ(
-      "strict sanctions.  Your account holds information that is applied to "
-      "all\n\r");
-    writeToQ(
+      "person.\n\r",
+      "One account can hold multiple characters. Creating more than one "
+      "account\n\r",
+      "for yourself is a violation of ",
+      MUD_NAME,
+      " multiplay rules and will lead to\n\r",
+      "strict sanctions. Your account holds information that is applied to "
+      "all\n\r",
       "characters that you own (including: generic terminal type, time "
-      "zone\n\r");
-    writeToQ(
-      "you play from, etc.).  You are required to enter a valid and "
-      "unique\n\r");
-    writeToQ(
-      "E-mail address which will be kept secret and used by the "
-      "game-maintainers\n\r");
-    writeToQ(
-      "to inform you if a serious problem occurs with your account.\n\r\n\r");
-    writeToQ(
-      "Note that the only password protection is at the account level.  Do "
-      "not\n\r");
-    writeToQ(
+      "zone\n\r",
+      "you play from, etc.). You may provide a valid and unique\n\r",
+      "e-mail address which will be kept secret and used by the "
+      "game maintainers\n\r",
+      "to inform you if a serious problem occurs with your account.\n\r\n\r",
+      "Note that the only password protection is at the account level. Do "
+      "not\n\r",
       "reveal your password to others or they have access to ALL of your "
-      "characters.\n\r\n\r");
+      "characters.\n\r\n\r",
+    };
+
+    for (const auto& message : messages) {
+      writeToQ(message);
+    }
 
     output.push(CommPtr(
       new LoginComm("user", "Type NEW to generate a new account.\n\rLogin: ")));
     return FALSE;
   } else if (my_arg == "1") {
-    FILE* fp = fopen("txt/version", "r");
-    if (!fp) {
-      vlogf(LOG_FILE, "No version file found");
-    } else {
-      if (!fgets(buf, 79, fp))
-        vlogf(LOG_FILE, "Unexpected read error in txt/version");
-      // strip off the terminating newline char
-      buf[strlen(buf) - 1] = '\0';
+    static const int years_of_service = []() {
+      tm current_date;
+      time_t conv_time;
+      conv_time = time(&conv_time);
+      localtime_r(&conv_time, &current_date);
+      int years_of_service = current_date.tm_year;
+      if ((current_date.tm_mon < 4) ||
+          (current_date.tm_mon == 4 && current_date.tm_mday <= 5)) {
+        years_of_service -= 93;
+      } else {
+        years_of_service -= 92;
+      }
+      return years_of_service;
+    }();
```

Comment:
Why a lambda? If you want a scope, you could just use {}.

Reply:
Just used a self-executing anonymous lambda to initialize a static variable, so that the code only gets executed once per boot since that value won't change once it's been calculated.